libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -111,46 +96,84 @@ func makeReporter(beat beat.Info, cfg *common.Config) (report.Reporter, error) {
 		}), nil
 	}
 
-	monitoring := monitoring.Default.NewRegistry("xpack.monitoring")
+	params := map[string]string{}
+	for k, v := range defaultParams {
+		params[k] = v
+	}
+	for k, v := range config.Params {
+		params[k] = v
+	}
+
+	getPipeline := func(name string, interval time.Duration) (*pipeline.Pipeline, error) {
 
-	pipeline, err := pipeline.New(
-		beat,
-		monitoring,
-		queueFactory, out, pipeline.Settings{
-			WaitClose:     0,
-			WaitCloseMode: pipeline.NoWaitOnClose,
-		})
+		params["interval"] = interval.String()
+		out.Clients = nil
+
+		for _, host := range hosts {
+			client, err := makeClient(host, params, proxyURL, tlsConfig, &config)
+			if err != nil {
+				return nil, err
+			}
+			out.Clients = append(out.Clients, client)
+		}
+
+		monitoring := monitoring.Default.GetRegistry("xpack.monitoring." + name)
+
+		return pipeline.New(
+			beat,
+			monitoring,
+			queueFactory, out, pipeline.Settings{
+				WaitClose:     0,
+				WaitCloseMode: pipeline.NoWaitOnClose,
+			})
+	}
+
+	statsPipeline, err := getPipeline("stats", config.StatsPeriod)
```

Comment:
@urso This code works but is not very nice. The reason I have to define 2 different pipelines is because the interval params is different and the monitoring registry conflicts. I would prefer to have only 1 pipeline but be able to set the params on request. Are there other options?